store/src/java/com/zimbra/cs/html/OwaspHtmlSanitizer.java

```diff
@@ -1,36 +1,409 @@
+                // quoted text
+                showHideQuotedText = "gmail_quote".equals(value) || "yahoo_quoted".equals(value);
+            }
+            // check if the id attribute is listed
+            final int idIndex = attrs.indexOf("id");
+            if (idIndex >= 0) {
+                // remove the id attribute and its value
+                final String value = attrs.remove(idIndex + 1);
+                attrs.remove(idIndex);
+                // AOL uses a specific id value to indicate quoted text
+                showHideQuotedText = value.startsWith("AOLMsgPart");
+            }
+            // insert a class attribute with a value of "elided-text" to
+            // hide/show quoted text
+            if (showHideQuotedText) {
+                attrs.add("class");
+                attrs.add("elided-text");
+            }
+            return "div";
+        }
+    };
+
+    private static AttributePolicy IMG_SRC_NEUTER = new AttributePolicy() {
+
+        public String apply(String elementName, String attributeName, String srcValue) {
+            if (VALID_EXT_URL.matcher(srcValue).find() || (!VALID_INT_IMG.matcher(srcValue).find()
+                && !VALID_IMG_FILE.matcher(srcValue).find())) {
+                return null;
+            } else if (!VALID_INT_IMG.matcher(srcValue).find()
+                && VALID_IMG_FILE.matcher(srcValue).find()
+                && !VALID_CONVERTD_FILE.matcher(srcValue).find()) {
+                return null;
+            }
+            return srcValue;
+        }
+    };
+
+    private static AttributePolicy BG_NEUTER = new AttributePolicy() {
+
+        public String apply(String elementName, String attributeName, String srcValue) {
+            return null;
+        }
+    };
+
+    private static final AttributePolicy NO_SPACE_ENCODED_CHARS = new AttributePolicy() {
+
+        public String apply(String elementName, String attributeName, String value) {
+            value = removeAnySpacesAndEncodedChars(value);
+            value = AV_JS_ENTITY.matcher(value).replaceAll("JS-ENTITY-BLOCKED");
+            value = AV_SCRIPT_TAG.matcher(value).replaceAll("SCRIPT-TAG-BLOCKED");
+            return value;
+        }
+    };
+
+    private AttributePolicy BLOCK_FORM_SAME_HOST_POST_REQ = new AttributePolicy() {
+
+        public String apply(String elementName, String attributeName, String value) {
+            if (sameHostFormPostCheck == true && reqVirtualHost != null) {
+                try {
+                    URL url = new URL(value);
+                    String formActionHost = url.getHost().toLowerCase();
+
+                    if (formActionHost.equalsIgnoreCase(reqVirtualHost)) {
+                        value = value.replace(formActionHost, "SAMEHOSTFORMPOST-BLOCKED");
+                    }
+                } catch (MalformedURLException e) {
+                    ZimbraLog.soap
+                        .warn("Failure while trying to block mailicious code. Check for URL "
```

Comment:
typo: mailicious > malicious
nit: extra space - trailing space here and leading space on the next line.